### PR TITLE
Re-enable cosmosdb

### LINKED ISSUE #512: Re-enable CosmosDB module

### PR DESCRIPTION
**Pull request by jeremyrickard**
This PR introduces the needed refactoring to re-enable the cosmosdb module. 

Closes #512

### CODE REVIEW

**Comment by jeremyrickard** [comment]
Doing more e2e with parameters before I declare this ready to merge.

File: pkg/services/cosmosdb/catalog.go [hunk 1]
```diff
@@ -215,19 +213,19 @@ func (m *module) GetCatalog() (service.Catalog, error) {
 						"Table",
 					},
 				},
-				m.graphAccountManager,
-				service.NewPlan(&service.PlanProperties{
+				m.tableAccountManager,
```
Comment (jeremyrickard) [inline pkg/services/cosmosdb/catalog.go]: this was using the wrong service manager! manual e2e uncovered that

File: pkg/services/cosmosdb/common_types.go [hunk 2]
```diff
@@ -1,46 +1,14 @@
...
 	DatabaseAccountName      string `json:"name"`
 	FullyQualifiedDomainName string `json:"fullyQualifiedDomainName"`
 	IPFilters                string `json:"ipFilters"`
-}
-
-type cosmosdbSecureInstanceDetails struct {
-	ConnectionString string `json:"connectionString"`
-	PrimaryKey       string `json:"primaryKey"`
+	ConnectionString         string `json:"connectionString"`
+	PrimaryKey               string `json:"primaryKey"`
```
Comment (krancour) [inline pkg/services/cosmosdb/common_types.go]: @jeremyrickard shouldn't these two fields be of type `service.SecureString`?

Comment (jeremyrickard) [reply pkg/services/cosmosdb/common_types.go]: Absolutely should, good catch! Updating.

**Pull request merged**